lib/models/recruiting-pipeline/restage-orphaned-job-applications.js

```diff
@@ -0,0 +1,71 @@
+const { map } = require('lodash');
+
+module.exports = event => {
+  return async function (pipeline, next) {
+    let JobApplication = require('../../models/job-application'),
+        Pipeline = require('../../models/recruiting-pipeline'),
+        previousStages = [],
+        currentStages = [],
+        jobApplicationQuery = {};
+
+    if (pipeline.jobOpenings.length < 1) {
+      //editing the company default pipeline
+      jobApplicationQuery.stage = { $in: map(pipeline.stagesWas, '_id') };
+    }
+
+    if (pipeline.jobOpenings.length > 0) {
+      //editing job opening specific pipeline
+      jobApplicationQuery.jobOpening = { $in: pipeline.jobOpenings };
+
+      if (event === 'save') {
+        //find apps with a stage that doesnt match current pipeline stages
+        //if event is removal, we know the entire array of jobOpenings are orphans, no need to query for stages
+        jobApplicationQuery.stage = { $nin: map(pipeline.stages, '_id') };
+      }
+    }
+
+    let orphanedApplications = await JobApplication.find(jobApplicationQuery);
+
+    if (orphanedApplications.length < 1) {
+      next();
```

Comment:
```suggestion
      return next();
```
Right? Otherwise, it'll run the rest of this code.